doc/architecture/decisions/0011-control-applicant-access-using-delius-team-caseload.md

```diff
@@ -0,0 +1,168 @@
+
+#### Workflow manager
+**Role**: `WORKFLOW_MANAGER`   
+**Description**: CRU staff ensuring that assessment and allocation tasks are completed  
+**Authorised to**:
+- view all applications and their allocations to staff members
+- reassign applications to different staff members as necessary
+- mark application as withdrawn (e.g. if person has died)
+- ask applicant for further info, independent of the assessment itself (in MVP this is 
+  limited to adding notes based on off-line communication)
+
+#### Assessor
+**Role**: `ASSESSOR`  
+**Description**: Staff member who is tasked with assessing specific applications  
+**Authorised to**:
+- assess a particular application
+- add additional assessment content (recommendations, clarifications etc) to an application
+
+#### Matcher
+**Role**: `MATCHER`  
+**Description**: Staff member who is tasked with finding and booking suitable accommodation 
+   for an available time slot  
+**Authorised to**:
+- search for accommodation for a given application
+- book accommodation for the given application
+- edit booking 
+- escalate a task where it wasn't possible to find a match, but the priority is very high 
+
+#### Manager
+**Role**: `MANAGER`  
+**Description**: AP staff member or area manager  
+**Authorised to**:
+- record arrival
+- record non-arrival
+- extend a booking
+- view or cancel a booking
+- view / edit / create lost beds
+- view capacity (occupancy)
+- record departure
+- view residents
+
+#### Admin
+**Role**: `ROLE_ADMIN`  
+**Description**: Manages AP role  
+**Authorised to**:
+- add AP roles to Delius users
+- remove AP roles from Delius users
+
+### The "applicant role"
+
+An "applicant" needs access to more than their individual caseload, in order to create and
+access applications for people managed by fellow team members.
+
+We know that probation officers should have membership of only one team, but currently
+this is not the case. At present, approx 270 staff in the North East region are members of multiple teams.
+
+When restricting our "Applicant" role to CRNs which are within the caseload of their
+team(s) we are considering the following scenarios:
+
+1. applicant creates a new application for a given CRN 
+2. applicant views existing applications, created by themselves 
+3. applicant views existing applications, created by team members
+
+
+## Decision
+
+We're basing our applicant access control on the caseload(s) of the team(s) in which the
+applicant has membership.
+
+### 1. Creating an application
+
+To achieve [1] we're asking the probation-integration (PI) team to give us a new endpoint on the
+[approved-premises-and-delius](https://github.com/ministryofjustice/hmpps-probation-integration-services/tree/main/projects/approved-premises-and-delius) API which they've built.
+
+For example, something like:
+
+```sh
+curl "https://approved-premises-and-delius/{crn}/managingTeams/{delius-staffcode}"
```

Comment:
I feel like a more RESTful approach would be something like:

https://approved-premises-and-delius/teams/search

With a JSON payload eg:

```json
{ "crn": "{crn}", "staffCode": "{delius-staffcode}" }
```

Though I guess the implementation is up to the integrations team

Reply:
Another option would be to use query parameters as filters:
```
GET /teams?managesCrn={crn}&hasStaffMember={delius-staffcode}
{
    "teams": ["N01ABC"]
}
```

But we're happy to use whichever implementation - does your team have a preference?

Reply:
Thanks @marcus-bcl 

@brownbearsoftware Would our OpenAPI code generator have trouble with any of these approaches? 

If not then probably it makes sense for you to decide on a design which is consistent within your mono-repo?

Reply:
We'd be consuming this endpoint so it wouldn't need to touch OpenAPI Generate so any approach will work for us 👍

Reply:
good point! I forget which side of the fence I'm on!

Reply:
We ended up going with:
```
GET /teams/managingCase/{crn}?staffCode={staffCode}
{
  "teamCodes": ["N01ABC"]
}
```

Hope that works for you.  The endpoint is up for review now, so should be available to test soon.

Reply:
this looks good, thanks